internal/kgateway/utils/sanitize.go

```diff
@@ -23,3 +24,59 @@ func SanitizeForEnvoy(ctx context.Context, resourceName, resourceTypeName string
 	}
 	return resourceName
 }
+
+// sanitizeCookieName ensures the cookie name is valid per RFC 6265 and Gateway API conventions.
+// - Only ASCII, no control chars, no separators, not empty, not too long (max 64 chars recommended)
+// - Replaces illegal characters with '_'
+func SanitizeCookieName(name string) string {
+	if len(name) == 0 {
+		return "cookie"
+	}
+	// RFC 2616 separators: ()<>@,;:\"/[]?={} and whitespace
+	/*
+
+		https://www.rfc-editor.org/rfc/rfc2616#section-2.2
+		       token          = 1*<any CHAR except CTLs or separators>
+		       separators     = "(" | ")" | "<" | ">" | "@"
+		                      | "," | ";" | ":" | "\" | <">
+		                      | "/" | "[" | "]" | "?" | "="
+		                      | "{" | "}" | SP | HT
+
+	*/
+	separators := "()<>@,;:\\\"/[]?={} \t"
+
+	return strings.Map(func(c rune) rune {
+		if unicode.IsControl(c) || c > unicode.MaxASCII || strings.ContainsRune(separators, c) {
+			return '_'
+		}
+		return c
+	}, name)
+}
+
+// sanitizeHeaderName ensures the header name is valid per RFC 7230 and Gateway API conventions.
```

Comment:
nit: Match the function name.

```suggestion
// SanitizeHeaderName ensures the header name is valid per RFC 7230 and Gateway API conventions.
```